pkg/resources/resource_database.go

```diff
@@ -101,6 +122,12 @@ func databaseCreate(ctx context.Context, d *schema.ResourceData, meta interface{
 	if err != nil {
 		return diag.FromErr(err)
 	}
+
+	// set id with the region
```

Comment:
So we have to set `TransformIdWithRegion` in both the create and read for state imports and the initial migration?

Reply:
Yes, I _think_ that this is the case. We should use `TransformIdWithRegion` in both the create and read functions as it ensures consistency in how IDs are managed and stored in the Terraform state for new resources.